aten/src/ATen/native/cuda/ReduceSumProdKernel.cu

```diff
@@ -57,8 +57,32 @@ struct nansum_functor {
   }
 };
 
-constexpr char prod_name[] = "prod";
+// jiterated specialization for `complex<Half>`
```

Comment:
I think this specialisation should be for all complex types. With that adding complex support will not negatively affect the AOT build-time.

Reply:
Sure, updated the PR! Thanks, Kshitij!